compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6112,7 +6120,25 @@ private BType checkFieldAccessExpr(BLangFieldBasedAccess fieldAccessExpr, BType
             if (fieldAccessExpr.fieldKind == FieldKind.WITH_NS) {
                 resolveXMLNamespace((BLangFieldBasedAccess.BLangNSPrefixedFieldBasedAccess) fieldAccessExpr);
             }
-            BType laxFieldAccessType = getLaxFieldAccessType(varRefType);
+
+            BType laxFieldAccessType;
+            if (fieldAccessExpr.isWithCheckExpr && fieldAccessExpr.expectedType.tag != TypeTags.NONE) {
+                BUnionType unionType = (BUnionType) fieldAccessExpr.expectedType;
+                // Filter out the list of types which are not equivalent with the error type.
+                List<BType> expectedTypes = unionType.getMemberTypes().stream()
+                        .filter(type -> !symTable.errorType.equals(type)).collect(Collectors.toList());
```

Comment:
Don't we have to ignore all error types? If so we need to check by tag.
```suggestion
                        .filter(type -> type.tag != TypeTags.ERROR).collect(Collectors.toList());
```

Reply:
Even if we were only checking for `symTable.errorType`, we can just do 
```suggestion
                        .filter(type -> symTable.errorType != type).collect(Collectors.toList());
```

instead of `equals()`.

Reply:
Changed.